pennylane/fermi/conversion.py

```diff
@@ -148,3 +148,139 @@ def _(fermi_operator: FermiSentence, ps=False, wire_map=None, tol=None):
+    where :math:`X`, :math:`Y`, and :math:`Z` are the Pauli operators and :math:`N` is the number of qubits/spin orbitals.
+
+    Args:
+        fermi_operator(FermiWord, FermiSentence): the fermionic operator
+        n_qubits (int): Number of qubits/spin orbitals in the system
+        ps (bool): whether to return the result as a PauliSentence instead of an
+            Operator. Defaults to False.
+        wire_map (dict): a dictionary defining how to map the oribitals of
+            the Fermi operator to qubit wires. If None, the integers used to
+            order the orbitals will be used as wire labels. Defaults to None.
+        tol (float): tolerance for discarding the imaginary part of the coefficients
+
+    Returns:
+        Union[PauliSentence, Operator]: a linear combination of qubit operators
+
+    **Example**
+
+    >>> w = FermiWord({(0, 0) : '+', (1, 1) : '-'})
+    >>> parity_transform(w, n_qubits=6)
+    (-0.25j*(PauliY(wires=[0]))) + ((-0.25+0j)*(PauliX(wires=[0]) @ PauliZ(wires=[1]))) +
+    ((0.25+0j)*(PauliX(wires=[0]))) + (0.25j*(PauliY(wires=[0]) @ PauliZ(wires=[1])))
+
+    >>> parity_transform(w, n_qubits=6, ps=True)
+    -0.25j * Y(0)
+    + (-0.25+0j) * X(0) @ Z(1)
+    + (0.25+0j) * X(0)
+    + 0.25j * Y(0) @ Z(1)
+
+    >>> parity_transform(w, n_qubits=6, ps=True, wire_map={0: 2, 1: 3})
+    -0.25j * Y(2)
+    + (-0.25+0j) * X(2) @ Z(3)
+    + (0.25+0j) * X(2)
+    + 0.25j * Y(2) @ Z(3)
+    """
+    return _parity_transform_dispatch(fermi_operator, n_qubits, ps, wire_map, tol)
+
+
+@singledispatch
+def _parity_transform_dispatch(fermi_operator, n_qubits, ps, wire_map, tol):
+    """Dispatches to appropriate function if fermi_operator is a FermiWord or FermiSentence."""
+    raise ValueError(f"fermi_operator must be a FermiWord or FermiSentence, got: {fermi_operator}")
+
+
+@_parity_transform_dispatch.register
+def _(fermi_operator: FermiWord, n_qubits, ps=False, wire_map=None, tol=None):
+    wires = list(fermi_operator.wires) or [0]
+    identity_wire = wires[0]
+
+    if len(fermi_operator) == 0:
+        qubit_operator = PauliSentence({PauliWord({}): 1.0})
+
+    else:
+        coeffs = {"+": -0.5j, "-": 0.5j}
+        qubit_operator = PauliSentence({PauliWord({}): 1.0})  # Identity PS to multiply PSs with
+
+        for item in fermi_operator.items():
+            (_, wire), sign = item
+            if wire >= n_qubits:
+                raise ValueError(
+                    f"Can't create or annihilate a particle on qubit number {wire} for a system with only {n_qubits} qubits"
+                )
+
+            z_string = {wire - 1: "I"} if wire == 0 else {wire - 1: "Z"}
+            x_string = dict(zip(range(wire + 1, n_qubits), ["X"] * (n_qubits - wire)))
+
+            qubit_operator *= PauliSentence(
+                {
+                    PauliWord({**z_string, **{wire: "X"}, **x_string}): 0.5,
+                    PauliWord({**{wire: "Y"}, **x_string}): coeffs[sign],
+                }
+            )
```

Comment:
Nice try, but I think we can make it even simpler. Consider the following changes: 

1. For the if-else block, note that the `qubit_operator` is defined the same way in both cases. So it doesn't matter which of the two paths are executed, this variable is defined as the same. For this reason we can actually pull its definition out of the if else section completely.  
2. Also note, the for-loop which iterates over the fermi_operator. If the operator is empty (len == 0), then the for loop does not get executed. So we actually don't need the initial if-else check at all, since if the len(fermi_operator) == 0, then the loop already won't execute. 
3. I see what you were trying to do by adding the identity term for the z-string but it adds an unnecessary overhead since the PauliWord manually deletes any identity terms. Consider something like: 

```
x_string = dict(zip(range(wire + 1, n_qubits), ["X"] * (n_qubits - wire)))
pw1 = PauliWord({**{wire: "X"}, **x_string}) if wire == 0 else PauliWord({**{wire - 1: "Z"}, **{wire: "X"}, **x_string}) 
pw2 = PauliWord({**{wire: "Y"}, **x_string})

qubit_operator *= PauliSentence({pw1: 0.5, pw2: coeffs[sign]})
```

Reply:
For the second point, I don't think we can get rid of the if statement completely, we can change it slightly though. The for loop which iterates over the fermi_operator is only executed when fermi_operator is of type FermiSentence, but when the type is FermiWord, we need that if statement.

Reply:
I don't think I understand? This function will only act on FermiWords because of the single_dispatch. So we can safely assume that the type of fermi_operator is `FermiWord`. Now notice that FermiWord is implemented as a dictionary. An empty FermiWord would have no items, thus the for loop would be empty. Does that clear up the confusion? 

Which if statement are you referring to?

Reply:
oh I see what you mean now, sorry about the confusion.